src/NetMQ/Core/Patterns/Server.cs

```diff
@@ -186,7 +183,7 @@ protected override bool XRecv(ref Msg msg)
             if (!received)
                 return false;
 
-            Debug.Assert(pipe != null);
+            Assumes.NotNull(pipe);
```

Comment:
How smart is the compiler? Can he figure out that pipe cannot be null here? as you check the received boolean

Reply:
So actually the old code checked `pipe` against `null`, which showed the compiler that we thought the value might possibly be null, and so elevated the annotation on `pipe` from "not null" to "possibly null", which then led to warnings later on.

I converted a lot of `Debug.Assert(foo != null)` to `Assumes.NotNull(foo)` to get around these warnings. In this case (and the one above) it seems unlikely that `pipe` is ever null, so I'll remove both checks.